File: android/src/main/java/com/flutter_rust_bridge/xue_hua_media_compression/XueHuaMediaCompressionPlugin.java
package com.flutter_rust_bridge.xue_hua_media_compression;

import android.content.Context;
import androidx.annotation.NonNull;
import io.flutter.embedding.engine.plugins.FlutterPlugin;

public class XueHuaMediaCompressionPlugin implements FlutterPlugin {
    static {
        System.loadLibrary("xue_hua_media_compression");
    }

    private static native void initAndroid(Context ctx);

    @Override
    public void onAttachedToEngine(@NonNull FlutterPluginBinding binding) {
        initAndroid(binding.getApplicationContext());
    }

    @Override
    public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {}
}

File: android/src/main/kotlin/com/flutter_rust_bridge/xue_hua_media_compression/XueHuaMediaCompressionPlugin.kt
package com.flutter_rust_bridge.xue_hua_media_compression

import android.content.Context
import io.flutter.embedding.engine.plugins.FlutterPlugin

class XueHuaMediaCompressionPlugin : FlutterPlugin {
    companion object {
        init {
            System.loadLibrary("xue_hua_media_compression")
        }

        @JvmStatic
        external fun initAndroid(context: Context)
    }

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        initAndroid(binding.applicationContext)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}
